policyengine_us/variables/gov/states/va/tax/income/deductions/va_itemized_deductions.py

```diff
@@ -7,8 +7,54 @@ class va_itemized_deductions(Variable):
     label = "Virginia itemized deductions"
     unit = USD
     definition_period = YEAR
-
-    reference = (
-        "https://law.lis.virginia.gov/vacodefull/title58.1/chapter3/article2/"
-    )
+    reference = "https://www.tax.virginia.gov/sites/default/files/taxforms/individual-income-tax/2021/schedule-2021.pdf"
     defined_for = StateCode.VA
+
+    def formula(tax_unit, period, parameters):
+        # The federal limitation on itemized deductions does not apply during the TCJA period
+        # Virginia still applies the limitation
+        year = period.start.year
+        if year >= 2018 and year <= 2026:
+            instant_str = f"2017-01-01"
+        else:
+            instant_str = period
+        irs = parameters(instant_str).gov.irs.deductions.itemized.limitation
+        va = parameters(period).gov.states.va.tax.income.deductions.itemized
+
+        # va itemized deductions
+        itm_deds_less_salt = tax_unit("itemized_deductions_less_salt", period)
+        uncapped_property_taxes = add(tax_unit, period, ["real_estate_taxes"])
+        va_itm_deds = itm_deds_less_salt + uncapped_property_taxes
+
+        # Part A: If AGI from federal return is over a certain amount, then
+        # limitations to the itemized deduction are applied
+        federal_agi = tax_unit("adjusted_gross_income", period)
+        filing_status = tax_unit("filing_status", period)
+        applicable_amount = va.applicable_amount[filing_status]
+        agi_adjustment = irs.agi_rate * max_(
```

Comment:
or some more descriptive term than `excess` - to keep statements readable with one operation each
```suggestion
        excess = max_(federal_agi - applicable_amount, 0)
        agi_adjustment = irs.agi_rate * excess
```